jsapp/xlform/src/mv.validationLogicHelpers.coffee

```diff
@@ -98,11 +98,16 @@ module.exports = do ->
         <div class="card__settings__fields__field">
           <label for="#{@context.helper_factory.current_question.cid}-handcode">#{t("Validation Code:")}</label>
           <span class="settings__input">
-            <input type="text" name="constraint" id="#{@context.helper_factory.current_question.cid}-handcode" class="text" value="#{@criteria}">
+            <input type="text" name="constraint" id="#{@context.helper_factory.current_question.cid}-handcode" class="text" value="">
           </span>
         </div>
       """)
       @textarea = @$handCode.find('#' + @context.helper_factory.current_question.cid + '-handcode')
+      # Textarea starts with empty value - setting it in HTML caused error with
+      # mangled values (due to " character). Set initial value here, so all the
+      # special characters are saved.
+      @textarea.val(@criteria)
```

Comment:
Curious what is the difference between `value=` in the HTML element vs javascript value setting?

Reply:
The problem is not with the `value=` itself, but by the fact that the whole HTML code for this is being set with this long string (`"""`) and there JS is first putting the whole string together, and when it gets to `value="regex"with"quotes"` it recognizes `value="regex"` as attribute with value and then the `with"quotes"` is being ommited as not a valid attribute

Reply:
When using JS to set value, we are operating on JS HTML object, and the value setting functions have a safe way of storing `"` :)